modules/homa-lite/src/lib.rs

```diff
@@ -194,13 +224,31 @@ pub mod module {
 				Error::<T>::ExceededStakingCurrencyMintCap
 			);
 
-			let current_batch = Self::current_batch();
 			let staking_currency = T::StakingCurrencyId::get();
 
 			// ensure the user has enough funds on their account.
 			T::Currency::ensure_can_withdraw(staking_currency, &who, amount)?;
 
-			// Cross-chain transfers the staking assets.
+			// Calculate how much Liquid currency is to be minted.
+			// Gets the current exchange rate
+			let staking_total = Self::staking_total_issuance();
+			let exchange_rate = match staking_total.is_zero() {
```

Comment:
`Ratio::checked_from_rational(liquid_total, staking_total).unwrap_or_else(T::DefaultExchangeRate::get)`

Reply:
this is brilliant